.github/workflows/python-tests.yml

```diff
@@ -14,7 +14,9 @@ jobs:
       with:
         python-version: 3.8
     - name: Install apt dependencies
-      run: sudo apt-get install libdbus-1-dev libdbus-glib-1-dev python3-dev
+      run: sudo apt-get install libdbus-1-dev libdbus-glib-1-dev python3-dev gpg
+    - name: Import gpg key
+      run: gpg --import keys/key-file-pub.gpg
```

Comment:
Tests should use a special key. We don't want to have to update all tests if we rotate the key for some business reason.

Reply:
Yes @posterzh - as a general rule we try to avoid using actual input files for tests. In this case you can use gpg to [create a new key pair](https://github.com/NebraLtd/Hotspot-Production-Tool/tree/kashifpk/137-shipping-destinations#create-key-pair) during test setup. Or just generate one and store it with test cases that will only be used for tests.

This is going to be a bit tricky because python-gpg library used the actual gpg binary which assumes the default home folder (i.e, --homedir ~/.gnupg). Ideally you'd want to set the gpg homedir before running tests so it uses the keys you have either generated as part of test setup or already stored with tests. Since the library already supports this by creating the gpg object like:

```python
gpg = gnupg.GPG(gnupghome='/path/to/home/directory')
```

you can use this for testing assuming you update your code [here](https://github.com/NebraLtd/hm-diag/blob/126d57ed823dd3078269d40caeecdfb4e8e101ca/hw_diag/utilities/security.py#L3) to also accept a homedir path which if not present should default to `~/.gnupg`. Also another note, the gpg object you create should probably be created inside of the function.

Reply:
I agree to use gpg test key for test cases. I think it can be done without using custom homedir.